ipatests/test_integration/test_ipa_idrange_fix.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (C) 2024  FreeIPA Contributors see COPYING for license
+#
+
+"""
+Module provides tests for ipa-idrange-fix CLI.
+"""
+
+import logging
+import re
+
+from ipatests.pytest_ipa.integration import tasks
+from ipatests.test_integration.base import IntegrationTest
+
+
+logger = logging.getLogger(__name__)
+
+
+class TestIpaIdrangeFix(IntegrationTest):
+    @classmethod
+    def install(cls, mh):
+        super(TestIpaIdrangeFix, cls).install(mh)
+        tasks.kinit_admin(cls.master)
+
+    def test_no_issues(self):
+        """Test ipa-idrange-fix command with no issues."""
+        result = self.master.run_command(["ipa-idrange-fix", "--unattended"])
+
+        expected_under1000 = "No IDs under 1000 found"
+        expected_nochanges = "No changes proposed for existing ranges"
+        expected_newrange = "No new ranges proposed"
+        expected_noissues = "No changes proposed, nothing to do."
+        assert expected_under1000 in result.stderr_text
+        assert expected_nochanges in result.stderr_text
+        assert expected_newrange in result.stderr_text
+        assert expected_noissues in result.stderr_text
+
+    def test_idrange_no_rid_bases(self):
+        """Test ipa-idrange-fix command with IDrange with no RID bases."""
+        self.master.run_command([
+            "ipa",
+            "idrange-add",
+            "idrange_no_rid_bases",
+            "--base-id", '10000',
+            "--range-size", '20000',
+        ])
+
+        result = self.master.run_command(["ipa-idrange-fix", "--unattended"])
+        expected_text = 'RID bases updated for range idrange_no_rid_bases'
+        assert expected_text in result.stderr_text
+
+        # Remove IDrange with no rid bases
+        self.master.run_command(["ipa", "idrange-del", "idrange_no_rid_bases"])
+
+    def test_users_outofrange(self):
+        """Test ipa-idrange-fix command with users out of range."""
+        for i in range(1, 20):
+            self.master.run_command([
+                "ipa",
+                "user-add",
+                "testuser{}".format(i),
+                "--first", "Test",
+                "--last", "User {}".format(i),
+                "--uid", str(100000 + i * 10),
+            ])
+
+        result = self.master.run_command(["ipa-idrange-fix", "--unattended"])
+        expected_text = r"Range [\w\.]+_id_range_\d{3} created successfully"
+        match = re.search(expected_text, result.stderr_text)
+        assert match is not None
+
+        # Remove users out of range and created IDrange
+        for i in range(1, 20):
+            self.master.run_command([
+                "ipa",
+                "user-del",
+                "testuser{}".format(i)
+            ])
+        if match is not None:
+            self.master.run_command([
+                "ipa",
+                "idrange-del",
+                match.group(0).split(" ")[1]
+            ])
```

Comment:
What range is this removing? The one from the previous test? Why regex it?

Reply:
We're removing range we created inside this test for these 20 out-of-range users. Tool auto-detects the range name - it will look like $REALM_id_range_$COUNTER, so I've decided to use regex not to hardcode any of those values. Match finds phrase "Range $RANGENAME created successfully", so I just take $RANGENAME here to make sure we delete only the range we created for the test.